test/ERC4626.t.sol

```diff
@@ -159,6 +166,116 @@ contract SparkVaultERC4626Test is SparkVaultTestBase {
         vault.previewWithdraw(assets);
     }
 
+    function test_maxDeposit_depositCapLeTotalAssets() public {
+        // Deposit cap is currently 2_100_000e6 and total assets is slightly above 1_000_000e6
+        vm.prank(admin);
+        vault.setDepositCap(1_000_107.459782e6 - 1);
+        uint256 maxDepositUser1 = vault.maxDeposit(user1);
+        uint256 maxDepositUser2 = vault.maxDeposit(user2);
+        assertEq(maxDepositUser1, 0);
+        assertEq(maxDepositUser2, maxDepositUser1);
+
+        vm.prank(admin);
+        vault.setDepositCap(1_000_107.459782e6);
+        maxDepositUser1 = vault.maxDeposit(user1);
+        maxDepositUser2 = vault.maxDeposit(user2);
+        assertEq(maxDepositUser1, 0);
+        assertEq(maxDepositUser2, maxDepositUser1);
+
+        vm.prank(admin);
+        vault.setDepositCap(1_000_107.459782e6 + 1);
+        maxDepositUser1 = vault.maxDeposit(user1);
+        maxDepositUser2 = vault.maxDeposit(user2);
+        assertEq(maxDepositUser1, 1);
+        assertEq(maxDepositUser2, maxDepositUser1);
+    }
+
+    function test_maxDeposit_depositCapGtTotalAssets() public {
+        // Deposit cap is currently 2_100_000e6 and total assets is slightly above 1_000_000e6
+        uint256 maxDepositUser1 = vault.maxDeposit(user1);
+        uint256 maxDepositUser2 = vault.maxDeposit(user2);
+
+        assertEq(maxDepositUser1, 1_099_892.540218e6);
+        assertEq(maxDepositUser2, maxDepositUser1);
+
+        assertEq(maxDepositUser1, vault.depositCap() - vault.totalAssets());
+    }
+
+    function test_maxMint_depositCapLeTotalAssets() public {
+        // Deposit cap is currently 2_100_000e6 and total assets is slightly above 1_000_000e6
+        assertEq(vault.totalAssets(), 1_000_107.459782e6);
+
+        vm.prank(admin);
+        vault.setDepositCap(1_000_107.459782e6 - 1);
```

Comment:
Same changes as in deposit, use totalAssets - 1, use 0 and 1 in assertions, and put everything on new lines